### ISSUE TITLE
Project starter template

### ISSUE THREAD

**Issue by wbashir** [techpines/bone.io#15]
Great framework. Just curios to see if there is a scaffolded template to start a new project with?

**Comment by techpines** [comment]
@warsamebashir Not right now, but I think that's a great idea!  

The only tricky bit is the templating engine.  I don't want to endorse a client side templating engine, (like handlebars, underscore or jade), but I might be able to put something together using https://github.com/visionmedia/consolidate.js/

**Comment by lorefnon** [comment]
@techpines I don't believe @warsamebashir was talking about a templating engine like handlebars. I think he war referring to something similar to https://github.com/angular/angular-seed ie. a minimal application skeleton which people can fork/download and start building upon. 

I have been using bone.io for a hobby project and find its simplistic approach very appealing. Since there is hardly any boilerplate involved, having starter seed skeletons is not really necessary. But nevertheless, if you would like to advocate a particular directory structure or way of organizing files - a scaffold can come in handy. 

I am not sure if you would like to move towards a more opinionated structuring, but in case you do - I would like to contribute towards command line generators. Coming from a rails background, I found found scaffold generators to be very helpful for rapid development.

**Comment by wbashir** [comment]
I think the opinionated folder structure, examples of particular template engines and backend persistence is what i was looking for.

**Comment by techpines** [comment]
@lorefnon I like the idea of a bone.io-seed project.  The only reason I mentioned "templating" is that if you make a seed project, I think you're going to have to make some decisions, like which templating engine to use.  

As a reference, here is the directory structure I usually start out with:

```
app.js       --> entry point for the node app
io.js        --> bone.io server routes
browser/     --> browser code
    app.js   --> entry point for the browser app (router/configuration)
    io.js    --> bone.io browser routes
    views.js --> bone.io view declarations
templates/   --> client and server side templates 
static/      --> static content
style/       --> css (maybe less, stylus)
gruntfile.js --> something to at least bundle all the client side assets together
package.json --> some base requires like bone.io, express, socket.io, etc
```

Obviously, that can change, if there is enough view code then `views.js` would become a folder called `views` with a bunch of `*.js` files in it.

Personally, I'm not always picking the technologies that we use for client projects, so it's helpful that we can just plug bone.io in to something that's already in place.  If it's up to me, I like to use Jade and Less, but my opinion isn't always as important as the team I'm working with.

If you build a "bone.io-seed" project, I'd be more than happy to link back to it in the docs.  I think it would definitely add value for people looking to get up and running quickly.

**Comment by techpines** [comment]
@warsamebashir Definitely more examples are coming.

**Comment by lorefnon** [comment]
@techpines  Thank you for your inputs. After exploring a couple of options, I found that a popular community project Yeoman provided a lot more flexibility  as compared to a static seed project. I prototyped [a simple Yeoman generator](https://github.com/lorefnon/generator-bone) that provides a CLI based interactive wizard like interface to guide the user through available options.
![yeoman1](https://f.cloud.github.com/assets/1449492/905499/9a43fc42-fc2f-11e2-9d66-a218b4c71d57.png)
![yeoman2](https://f.cloud.github.com/assets/1449492/905500/9d82bb6e-fc2f-11e2-88e3-b667f505c56c.png)

The concept is as of yet half baked and just creates a directory structure and a rudimentary home page, however, I plan to expand upon this. 

The directory structure is pretty much same as what you prescribed with only minor differences:

![yeoman3](https://f.cloud.github.com/assets/1449492/905509/43a07bee-fc30-11e2-847a-377f96a322b4.png)

If a simple seed project is desired for simplicity, I can just push a special case of generated project scaffold in a repository of its own.

Any criticism / feedback / suggestions are welcome. I will update on this thread once this is stable and backed by a decent test bench.

**Comment by ChrisMissal** [comment]
:+1:

**Comment by techpines** [comment]
Wow, this looks awesome! 

I can't seem to run this step, though:

``` bash
npm install -g generator-bone
```

Maybe you didn't publish, or maybe it's because of the problems npm seems to be having?

**Comment by lorefnon** [comment]
I haven't yet published the gem - wanted to put in a bit more effort into it before releasing.

**Comment by wbashir** [comment]
@lorefnon any word on this gem ?
